### ISSUE TITLE
REAPER crashes while Skype is running

### ISSUE THREAD

**Issue by pannal** [eiz/SynchronousAudioRouter#34]
Skype seems to have a firm grip on the virtual audio devices of SAR.
When it is running (no audio being transmitted) and I close REAPER, REAPER crashes after reopening.

Unless I kill Skype completely, I can't open REAPER.

I don't know whether this is fixable by SAR, though.

**Comment by pannal** [comment]
This also seems to happen with other apps - half of the time, when I run REAPER after boot, it crashes on load, then an unkillable reaper.exe lingers in task manager.

**Comment by pannal** [comment]
Disregard my last message - the lingering REAPER processes are due to the Nahimic audio portions of my ASUS Realtek onboard sound driver. Once I replaced them with the official ones from the Windows Update Catalog, the DLLs vanished and the bug as well.

The original issue still persists, though, I'm afraid.

**Comment by dreamcat4** [comment]
@pannal as another user who experienced the same type of problems here: I really do want this project to succeed, it just isn't there yet in terms of stability. In the meantime I have been able to install Jack2 on windows and use the utility program 'qjackctrl.exe' to configure and run the jackd server successfully. 

When coupled with 'VoiceMeeter Banana' (for non-ASIO programs), and ASIO4All, they can do all of the necessary routing / patching instead. Albeit with the limitation of 96Khz maximum sample rate imposed by Voicemeeter Banana on all the shared ASIO bus. However it is rock solid.

Again, I do hope this project can be improved in future. Until then! Kind Regards.

**Comment by skaufhold** [comment]
Just to add to this: If you can get your hands on it, try the O Deus AsioLink Driver. It does exactly what this driver does, minus the automatic assignments of programs to WDM devices. But there are other tools that do that. Here's the catch: You can't actually buy AsioLink through the proper channels because its developer sadly passed away a while back.

That said, I do think once this driver gets more stable I'd probably prefer it to AsioLink because it only creates exactly the WDM devices I actually need and has the automatic WDM device assignments.  AsioLink creates a whole bunch (16 in/16 out) in advance.

Can't really fault the creator of this driver, it's just a side project for him and it would be unfair to demand of him to put all his time into fixing the bugs right now. Especially considering it's open source already. I'd try to fix some bugs, but developing Windows kernel drivers isn't exactly my area of expertise and just like the creator of this driver, my time is limited and spending weeks on learning the ins and outs of it isn't feasible for me right now.

**Comment by dreamcat4** [comment]
Hey @skaufhold I also tried the ASIOLink utility before moving over to Jack.... It had some issues for me on windows 10. This is where i dig up ny notes and try to remember what it was...

I believe it might have been those playback / recording devices named 'ADAVPRO'.... were causing ASIO4ALL to have a higher latency. Could not get below 512 samples. When I uninstalled them completely, then I could bring my latency down with jack to only 128 samples.

The other issue with it is the interface is duplicated, and rounting is sometimes too roundabout. Also you have to launch client applications in a particular order / or have to be relaunching them to have the ASIO4ALL driver appear on the instance for your DAW.

By using just only 'qjackctl.exe' in Jack2, it doesnt have any of those issues. Also I would like to mention that another utility 'ASIOBridge / Hifi Cable' looked promising. However simply does not work right on win10. It was last updated in 2013. Kind Regards.

**Comment by pannal** [comment]
Yeah I know about Voicemeeter Banana, VBCable and all the other good stuff and have used them in the past.

Sadly nothing compares to a working REAPER setup with ASIO4ALL and SAR in terms of ease of use and expandability. Also the performance is really great when coupled with high end microphone filters.

**Comment by 2L40K** [comment]
**skaufhold** thanks for mentioning O Deus AsioLink Driver.
But what are the other tools that do "automatic assignments of programs to WDM devices"?

**dreamcat4** but why do you need ASIO4ALL?
